src/ddscxx/src/org/eclipse/cyclonedds/core/policy/PolicyDelegate.cpp

```diff
@@ -876,6 +876,87 @@ void OwnershipDelegate::set_c_policy(dds_qos_t* qos) const
 }
 
 
+//==============================================================================
+
+DataRepresentationDelegate::DataRepresentationDelegate(const DataRepresentationDelegate& other)
+    : kinds_(other.kinds_)
+{
+}
+
+DataRepresentationDelegate::DataRepresentationDelegate(const std::vector<dds::core::policy::DataRepresentationKind::Type>& kinds)
+    : kinds_(kinds)
+{
+    this->check();
+}
+
+void DataRepresentationDelegate::kinds(const std::vector<dds::core::policy::DataRepresentationKind::Type> &kinds)
+{
+    kinds_ = kinds;
+}
+
+std::vector<dds::core::policy::DataRepresentationKind::Type> DataRepresentationDelegate::kinds() const
+{
+    return kinds_;
+}
+
+bool DataRepresentationDelegate::operator ==(const DataRepresentationDelegate& other) const
+{
+    return other.kinds() == kinds_;
+}
+
+void DataRepresentationDelegate::check() const
+{
+    /* The kind correctness is enforced by the compiler: nothing to check. */
+}
+
+void DataRepresentationDelegate::set_iso_policy(const dds_qos_t* qos)
+{
+    uint32_t n = 0;
+    dds_data_representation_id_t *kinds = NULL;
+    if (dds_qget_data_representation(qos, &n, &kinds) && n > 0 && kinds != NULL) {
+        kinds_.resize(n);
+        for (uint32_t i = 0; i < n; i++) {
+            switch (kinds[i]) {
+                case DDS_DATA_REPRESENTATION_XCDR1:
+                    kinds_[i] = dds::core::policy::DataRepresentationKind::XCDR1;
+                    break;
+                case DDS_DATA_REPRESENTATION_XML:
+                    kinds_[i] = dds::core::policy::DataRepresentationKind::XML;
+                    break;
+                case DDS_DATA_REPRESENTATION_XCDR2:
+                    kinds_[i] = dds::core::policy::DataRepresentationKind::XCDR2;
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    if (kinds)
+      dds_free(kinds);
+}
+
+void DataRepresentationDelegate::set_c_policy(dds_qos_t* qos) const
+{
+    const auto kinds = kinds_;
+    std::vector<dds_data_representation_id_t> reps;
+    for (const auto & k:kinds) {
+        switch (k) {
+        case dds::core::policy::DataRepresentationKind::XCDR1:
+            reps.push_back(static_cast<dds_data_representation_id_t>(DDS_DATA_REPRESENTATION_XCDR1));
+            break;
+        case dds::core::policy::DataRepresentationKind::XML:
+            reps.push_back(static_cast<dds_data_representation_id_t>(DDS_DATA_REPRESENTATION_XML));
+            break;
+        case dds::core::policy::DataRepresentationKind::XCDR2:
+            reps.push_back(static_cast<dds_data_representation_id_t>(DDS_DATA_REPRESENTATION_XCDR2));
+            break;
```

Comment:
Why use a switch statement to assign to values to one another? Both in C and in C++ the enum labels have explicitly set to the correct values, so you can safely assign by value here, casting the C++ enum to the C enum.

Reply:
Same as the previous comment.